src/backend/common/helpers/insights_notable_helper.py

```diff
@@ -0,0 +1,126 @@
+from typing import DefaultDict, Dict, List
+
+from backend.common.consts.award_type import AwardType
+from backend.common.consts.event_type import EventType
+from backend.common.helpers.insights_helper_utils import (
+    create_insight,
+    LeaderboardInsightArguments,
+    make_insights_from_functions,
+)
+from backend.common.models.insight import Insight, NotableEntry, NotablesData
+from backend.common.models.keys import EventKey, TeamKey, Year
+
+
+class InsightsNotableHelper:
+    @staticmethod
+    def make_insights(year: Year) -> List[Insight]:
+        return make_insights_from_functions(
+            year,
+            [
+                InsightsNotableHelper._calculate_notables_hall_of_fame,
+                InsightsNotableHelper._calculate_notables_world_champions,
+                InsightsNotableHelper._calculate_notables_division_winners,
+                InsightsNotableHelper._calculate_notables_division_finals_appearances,
+            ],
+        )
+
+    @staticmethod
+    def _create_notable_insight(
+        teams: Dict[TeamKey, List[EventKey]] | DefaultDict[TeamKey, List[EventKey]],
+        insight_type: int,
```

Comment:
Can we type this? (maybe even just like `InsightType: TypeAlias = int` in the insights type constants?